packages/grid/x-data-grid/src/hooks/core/useGridInitialization.ts

```diff
@@ -19,6 +21,9 @@ export const useGridInitialization = <
   props: DataGridProcessedProps,
 ) => {
   const privateApiRef = useGridApiInitialization<PrivateApi, Api>(inputApiRef, props);
+
+  useGridRefs(privateApiRef);
+  useGridTheme(privateApiRef);
```

Comment:
I have also lifted the `theme` in the grid state. This is done so that functions that select & operate on the state can be truly pure (in the real FP sense, not react "pure"). We should use a single mechanism for reactivity instead of storing part of the inputs in the `state` and other parts in react hooks (theme in `useContext`, etc).

This also allowed me to turn the visible pinned columns logic into a grid selector, which makes it way more resilient to change than the approach I used before, and simplifies the mental load required to understand that logic.